devito/jit_manager.py

```diff
@@ -15,7 +15,8 @@ class JitManager(object):
     _wrapped_functions = None
     COMPILER_OVERRIDE_VAR = "DEVITO_CC"
     REMOVE_FLAG_VAR = "DEVITO_REMOVE_FLAG"
-    _incompatible_flags = ["-Wshorten-64-to-32", "-Wstrict-prototypes", ("-arch", "i386")]
+    _incompatible_flags = ["-O2", "-Wshorten-64-to-32", "-Wstrict-prototypes", ("-arch", "i386")]
+    _compatible_flags = ["-O3", "-g"]
```

Comment:
This is fine for now, but we do definitely want to make our compiler flag handling more elegant soon.